frontend/src/Components/ReportDetails/ReportDetails.js

```diff
@@ -0,0 +1,267 @@
+                            let prevPage = this.state.pageNumber - 1;
+                            this.setState({pageNumber: prevPage});
+                            this.updateData({pageNumber: prevPage});    
+                        }
+                    }else{
+                        console.error(err);
+                    }
+                });
+    }
+
+    onChangePage = (pageNumber) => {
+        pageNumber += 1;
+        this.setState({
+            pageNumber : pageNumber
+        });
+        this.updateData({pageNumber});
+    }
+
+    onChangeRowsPerPage = (changeRowsPerPage) => {
+        this.setState({
+            pageNumber : 1,
+            itemsPerPage : changeRowsPerPage
+        })
+        this.updateData({pageNumber: 1, itemsPerPage: changeRowsPerPage});
+    }
+
+    showSnackbar = (type, message) => {
+        switch (type){
+            case "success":
+                this.setState({
+                    snackbarMessage : message,
+                    snackbarVariant : "success",
+                    openSnackbar : true
+                });
+                break;
+            case "error":
+                this.setState({
+                    snackbarMessage : message,
+                    snackbarVariant : "error",
+                    openSnackbar : true
+                });
+                break;
+            default:
+                console.error(`Snackbar type: '${type}' not supported!`)
+                break;
+        }
+    };
+
+    componentDidMount(){
+        if(this.state.reportId === undefined) return;
+        
+        this.updateData();        
+    }
+
+    setCheckedUnchecked = (row, value) => {
+        let {reportId, data} = this.state;
+        let supplyId = data[row].id;
+        let newValue = !value;
+        partialUpdateReportItem({reportId, supplyId, is_checked: newValue})
+            .then((res) => {
+                this.updateData();
+            }).catch((err) => {
+                console.error(err);
+                this.setState({
+                    openSnackbar: true,
+                    snackbarMessage: "Wystąpił błąd!",
+                    snackbarVariant: "error"
+                });
+            });
+    };
+
+    setIsCheckedSelected = (newValue) => {
+        let {reportId, data, rowsSelected} = this.state;
+
+        let allOk = true;
+        let someOk = false;
+        for (let key in rowsSelected) {
+            //Ommit API call if already checked
+            if(data[rowsSelected[key].index].is_checked == newValue) continue;
+
+            let supplyId = data[rowsSelected[key].index].id;
+            partialUpdateReportItem({reportId, supplyId, is_checked: newValue})
+                .then((res) => {
+                    someOk = true;
+                    this.updateData();
+                }).catch((err) => {
+                    console.error(err);
+                    allOk = false;
+                });
+        }
+
+        
+
+        if (allOk) {
```

Comment:
this may occur before updating items since partialUpdateReportItem is async function.